effekt/shared/src/main/scala/effekt/source/Tree.scala

```diff
@@ -697,8 +700,19 @@ object Resolvable {
     def resolve(using C: Context): symbols.CaptureSet = C.resolvedCapture(capt)
   }
 }
-export Resolvable.resolve
+export Resolvable.*
 
+extension [T](positioned: T) def sourceOfOpt(using C: Context): Option[String] = {
+  // XXX this should maybe be a method on `Source`?
+  def slice(range: kiama.util.Range): Option[String] = for {
+    start <- range.from.optOffset
+    end <- range.to.optOffset
+  } yield range.from.source.content.substring(start, end)
+
+  C.positions.getRange(positioned).flatMap(slice)
+}
+extension [T](positioned: T) def sourceOf(using C: Context): String =
+  positioned.sourceOfOpt.getOrElse { s"${positioned}" }
```

Comment:
Printing out the term as given by the user is something I'd really want. How can we legalise this? :)